tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -373,6 +374,8 @@ def f(scale, n_wires=10, offset=0.1, convert_to_hamiltonian=False):
         H = t1 + t2
         if convert_to_hamiltonian:
             H = H._pauli_rep.hamiltonian()  # pylint: disable=protected-access
+            H = qml.Hamiltonian(math.real(H.coeffs), H.ops)
```

Comment:
is this line necessary?

Reply:
yup! This is because when using `simulate`, the dtype of the terms in the hamiltonian don't get casted back so we end up with a complex expectation value. This doesn't play well with gradients since the standard gradient for expectation value only applies to scalar functions.

Reply:
This should probably be fixed in `devices.qubit.measure` then. We can cast to real if the imaginary part is zero.

Reply:
I agree, but I think that might be worth a separate PR.